docs/gitworkflow.rst

```diff
@@ -187,88 +193,66 @@ Developing your feature
 #. Consider adding regression tests for your feature in the test directory and
    also adding to the documentation for the code.
 
-#. If you want to share your development with others (or make a backup of your
-   repository in the cloud), upload the current status of your local feature
-   branch by pushing it to your personal repository::
-
-       git push --set-upstream origin some-new-feature
-
-   This also automatically connects the appropriate branch of your personal
-   repository on GitHub (`origin/some-new-feature`) with your local branch
-   (`some-new-feature`), so from now on, if you are on your `some-new-feature`
-   branch, a simple::
-
-       git push
+#. When the feature had been finished, clean up your commit history and squash
+   the intermediate commits into a few (e.g. one or two) relevant ones. Issue ::
 
-   command without any additional options will be enough to transfer your recent
-   changes on this branch to GitHub.
+     git rebase -i HEAD~N
 
+   where ``N`` should be replaced by the number of commits you had committed
+   into your feature branch so far. (You can find out the number by looking at
+   the git logs.) An interactive editor will pop up and you can select which
+   commits should be squashed together.
```

Comment:
I find this makes it quite involved with checking the log for the number of commits and I rarely use this in production.
If they should rebase against the default branch anyway, it would be simpler to use

    git rebase -i master